hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -782,81 +1047,155 @@ public Trigger copyTrigger(String tenantId, String triggerId, Map<String, String
-                newCondition = new StringCondition(newTrigger.getId(), c.getTriggerMode(),
-                        c.getConditionSetSize(), c.getConditionSetIndex(), dataIdMap.get(c.getDataId()),
-                        ((StringCondition)c).getOperator(), ((StringCondition)c).getPattern(),
-                        ((StringCondition)c).isIgnoreCase());
-            }
+            Condition newCondition = getMemberCondition(member, c, dataIdMap);
             if (newCondition != null) {
-                newCondition.setTenantId(newTrigger.getTenantId());
-                addCondition(newTrigger.getTenantId(), newTrigger.getId(), newCondition.getTriggerMode(), newCondition);
+                addCondition(newCondition);
             }
         }
 
         for (Dampening d : dampenings) {
-            Dampening newDampening = new Dampening(newTrigger.getId(), d.getTriggerMode(), d.getType(),
+            Dampening newDampening = new Dampening(member.getId(), d.getTriggerMode(), d.getType(),
                     d.getEvalTrueSetting(), d.getEvalTotalSetting(), d.getEvalTimeSetting());
-            newDampening.setTenantId(newTrigger.getTenantId());
+            newDampening.setTenantId(member.getTenantId());
+
+            addDampening(newDampening);
+        }
+
+        return member;
+    }
+
+    private Condition getMemberCondition(Trigger member, Condition groupCondition, Map<String, String> dataIdMap) {
+        Condition newCondition = null;
+        switch (groupCondition.getType()) {
+            case AVAILABILITY:
+                newCondition = new AvailabilityCondition(member.getId(), groupCondition.getTriggerMode(),
+                        groupCondition.getConditionSetSize(), groupCondition.getConditionSetIndex(),
+                        dataIdMap.get(groupCondition.getDataId()),
+                        ((AvailabilityCondition) groupCondition).getOperator());
+                break;
+            case COMPARE:
+                newCondition = new CompareCondition(member.getId(), groupCondition.getTriggerMode(),
+                        groupCondition.getConditionSetSize(), groupCondition.getConditionSetIndex(),
+                        dataIdMap.get(groupCondition.getDataId()),
+                        ((CompareCondition) groupCondition).getOperator(),
+                        ((CompareCondition) groupCondition).getData2Multiplier(),
+                        dataIdMap.get(((CompareCondition) groupCondition).getData2Id()));
+                break;
+            case EXTERNAL:
+                newCondition = new ExternalCondition(member.getId(), groupCondition.getTriggerMode(),
+                        groupCondition.getConditionSetSize(), groupCondition.getConditionSetIndex(),
+                        dataIdMap.get(groupCondition.getDataId()),
+                        ((ExternalCondition) groupCondition).getSystemId(),
+                        ((ExternalCondition) groupCondition).getExpression());
```

Comment:
Also for expressions, too.
As I think Expressions might be related to a specific metric, right ?
So, expanding that metric, might affect a multiple metrics.
I'm trying to imagine if we have a ExternalCondition for avg(ResponseTime) that wants to be applied into multiple machines too.
Perhaps it's a cornercase, but I put this note here in case it can be interesting to thing about this.